### PR TITLE
fix: update poetry & improve instructions

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

- Update Poetry to latest version.
- Improve instructions for local development.

## Related

No JIRA ticket.

## Changes

- **changed** Poetry version
- **changed** "Local Development" documentation
- **added** `package-mode = false` (to avoid warning)

## Testing

1. Follow [Local Development → via Python](https://github.com/DesignSafe-CI/DS-User-Guide/blob/fix/update-poetry-and-improve-instructions/README.md#a-via-python).
2. Verify you can serve documentation.
3. Follow [Local Development → via Docker](https://github.com/DesignSafe-CI/DS-User-Guide/blob/fix/update-poetry-and-improve-instructions/README.md#b-via-docker).
4. Verify you can serve documentation.

## UI

Skipped.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 name = "ds-user-guide"
 version = "0.1.0"
 description = ""
+package-mode = false
```
Comment (wesleyboar) [inline pyproject.toml:5]: This avoids a warning in newer version of Poetry, which user may install using "Local Development" → "via Docker", and is [accurate](https://python-poetry.org/docs/pyproject/#package-mode).

<details><summary>The Warning</summary>

> Installing the current project: ds-user-guide (0.1.0)
Warning: The current project could not be installed: No file/folder found for package ds-user-guide
If you do not want to install the current project use --no-root.
If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
In a future version of Poetry this warning will become an error!

</details>

File: Dockerfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ LABEL maintainer="TACC-ACI-WMA <wma_prtl@tacc.utexas.edu>"
 ARG DEBIAN_FRONTEND=noninteractive
 
 # https://python-poetry.org/docs/configuration/#using-environment-variables
-ENV POETRY_VERSION=1.4.2 \
+ENV POETRY_VERSION=1.8.2 \
```
Comment (wesleyboar) [inline Dockerfile:8]: If user follows "Local Development" → "via Python", they will install the latest Python. Those instructions force us to keep Docker's version up to date to ensure compatibility. Alternatively, I update instructions to install Poetry at a specific version.

Comment (silviamazzoni) [reply Dockerfile:8]: I would add a comment that you may not need the latest python version. it worked for my 10.13, afterall.

File: README.md
Comment (silviamazzoni) [inline README.md:1]: I would say:
Install dependencies (you need to do this only once) or something like this

looks great!

Comment (silviamazzoni) [reply README.md:1]: also,
5. Open the website at the URL returned in the last of the lines echoed by the program, e.g.
    http://127.0.0.1:8000/user-guide/

File: pyproject.toml (hunk 1 above)
Comment (silviamazzoni) [reply pyproject.toml:5]: poetry is lower case...

**Review by silviamazzoni — APPROVED** [review]
thank you for taking the time to do this!

File: Dockerfile (hunk 2 above)
Comment (wesleyboar) [reply Dockerfile:8]: Known supported versions are [>=3.10 and <3.12](https://github.com/DesignSafe-CI/DS-User-Guide/blob/fix/update-poetry-and-improve-instructions/pyproject.toml#L9). I've added note via a1f1c57.

File: README.md
Comment (wesleyboar) [reply README.md:1]: Thanks. Adjusted via 97c172e.

File: pyproject.toml (hunk 1 above)
Comment (wesleyboar) [reply pyproject.toml:5]: Unless one is pretentious enough to usurp a dictionary word as the name of their product, yes.

**Review by kks32 — APPROVED** [review]

**Comment by kks32** [comment]
Looks good! @wesleyboar

**Pull request merged**